apps/webapp/src/Replacements.tsx

```diff
@@ -0,0 +1,134 @@
+    start: "rounded-t-md md:rounded-l-md md:rounded-r-none",
+    end: "rounded-b-md md:rounded-r-md md:rounded-l-none",
+  }
+
+  const {
+    textColor,
+    bgColor,
+    highlightColor,
+    borderColor
+  } = styles[type];
+
+  const borderStyle = borderStyles[borderType];
+
+  const boxStyle = [textColor, bgColor, borderColor].join(" ");
+   
+  return (
+    <>
+      <div
+        className={`text-xs pl-2 order-1 md:order-1 md:col-span-1 md:row-start-1 md:row-end-2 ${textColor}`}
+      >
+        {label.toUpperCase()}
+      </div>
+      <div
+        className={
+          `px-2 py-2 h-full text-left font-medium border whitespace-pre-line
+          break-words order-2 md:order-3 md:col-span-1 md:row-start-2 md:row-end-3
+          ${boxStyle} ${borderStyle}`
+        }
+      >
+        <span className={`${highlightColor} p-1 box-decoration-clone`}>
+          {headline}
+        </span>
+      </div>
+    </>
+  )
+}
+
+function ReplacementItem({replacement}: ReplacementItemProps) {
+  return (
+    <Card>
+      <div
+        className="col-span-2 items-stretch px-2"
+      >
+        <div className="grid grid-cols-1 md:grid-cols-2 gap-2 md:gap-y-0">
+          <HeadlineContent
+            label="Original"
+            headline={replacement.originalHeadline}
+            type="original"
+            borderType="start"
+          />
+          <HeadlineContent
+            label="Replacement"
+            headline={replacement.replacementHeadline}
+            type="replacement"
+            borderType="end"
+          />
+          {/* Status and URL */}
+          <div className="flex flex-col sm:flex-row sm:items-center gap-2 my-2 text-sm text-gray-800 col-span-1 break-all order-5 md:order-5 md:col-span-2 md:row-start-3 md:row-end-4 md:my-2">
+            <div className="flex gap-2">
+              <button className="px-3 py-1 rounded-md bg-blue-500 text-white font-bold">Submit</button>
+              <span className="inline-block rounded-full bg-yellow-100 text-yellow-800 px-3 py-1 font-semibold text-sm shrink-0 text-left">{replacement.status}</span>
+            </div>
+            <a href={replacement.url} target="_blank" rel="noopener noreferrer" className="underline decoration-gray-400/50 hover:decoration-gray-800 transition-all text-left sm:px-1">{replacement.url}</a>
+          </div>
+        </div>
+      </div>
+    </Card>
+  );
+}
+
+export default function Replacements() {
+  const [sampleReplacementHeadlines, setSampleReplacementHeadlines] = useState<Replacement[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getSampleReplacementHeadlines()
+      .then((data) => setSampleReplacementHeadlines(data))
+      .catch((err) => {
+        console.error("Failed to fetch replacement headlines", err);
+        setSampleReplacementHeadlines([]);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  if (loading) {
+    return <Page><div>Loading...</div></Page>;
+  }
```

Comment:
This pattern is handled by [tanstack query](https://tanstack.com/query/latest). You would replace this with
```
const { data: smapleReplacementHeadlines } = useSuspenseQuery({
  queryKey: ["replacementHeadlines"],
  queryFn: getSampleReplacementHeadlines
]);
```
And that's it. While the query is processing it will bubble up to the `Suspense` boundary in the `QueryBoundaryProvider`. If there's an error it will bubble up to the `ErrorBoundary`.

Reply:
See `ParsedArticle` for an example